benchmarks/time/src/main/scala/strawman/collection/immutable/ArrayBaselineBenchmark.scala

```diff
@@ -0,0 +1,143 @@
+package strawman.collection.immutable
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+import strawman.collection.{ArrayView, View}
+import scala.{Any, AnyRef, Int, Long, Unit, Array}
+import scala.Predef.intWrapper
+
+@BenchmarkMode(scala.Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(1)
+@Warmup(iterations = 12)
+@Measurement(iterations = 12)
+@State(Scope.Benchmark)
+class ArrayBaselineBenchmark {
+
+  @Param(scala.Array("282", "73121", "7312102"))
+  var size: Int = _
+
+  @Param(scala.Array("39"))
+  var vLoSize: Int = _
+
+  var shortRangingFactor : Int = (size * 0.2).toInt
+
+  var v: Array[Long] = _
+  var vLo: Array[Long] = _
+
+  @Setup(Level.Trial)
+  def initData(): Unit = {
+
+    def fillArray(range: Int) = {
+      val array = new Array[Long](range)
+      var i = 0
+      while (i < range) {
+        array(i) = i.toLong
+        i += 1
+      }
+      array
+    }
+
+    v = fillArray(size)
+    vLo = fillArray(vLoSize)
+  }
+
+  @Benchmark
+  def sum(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      ret += v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquares(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      ret += v(i) * v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquaresEven(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      if (v(i) % 2L == 0L)
+        ret += v(i) * v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def filters(bh: Blackhole) = {
+    var i=0
+    var ret=0L
+    while (i < v.length) {
+      if (v(i) > 1 && v(i) > 2 && v(i) > 3 && v(i) > 4 && v(i) > 5 && v(i) > 6 && v(i) > 7)
```

Comment:
It is filtering indeed but I agree with your points about its exploitable structure. I would like to check fusion of multiple filters. What if I extract the values and blackhole the variables?